engine/src/main/java/io/camunda/zeebe/engine/state/appliers/VariableMigratedApplier.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.state.appliers;
+
+import io.camunda.zeebe.engine.state.TypedEventApplier;
+import io.camunda.zeebe.protocol.impl.record.value.variable.VariableRecord;
+import io.camunda.zeebe.protocol.record.intent.VariableIntent;
+
+final class VariableMigratedApplier implements TypedEventApplier<VariableIntent, VariableRecord> {
+
+  public VariableMigratedApplier() {}
+
+  @Override
+  public void applyState(final long key, final VariableRecord value) {
```

Comment:
Good question. Short answer: We do not have to update any data in the state for variables.

Let's look at the stored data of a variable:

https://github.com/camunda/zeebe/blob/7002d53a079c06ab3a94f5485f022681a41dc9ed/engine/src/main/java/io/camunda/zeebe/engine/state/variable/DbVariableState.java#L42-L44

> [!NOTE] 
> The VariableInstance contains the scope key and the value itself.

- The key of the variable does not change: it's the thing that uniquely identifies it
- The name does not change, we're just moving it to another process definition
- The value does not change, we're just moving it to another process definition
- The scope key does not change: The variable continues to exist at the same scope (i.e. element instance), it won't move to a different element instance. Instead, we've moved the element instance to a different process definition

So, why do we still append the event? Well, that's so that all downstream consumers know that the variable was migrated to a different process definition. And I think this makes sense because the `processDefinitionKey` and `bpmnProcessId` are properties of a Variable Record, although they are not stored in the state:

https://github.com/camunda/zeebe/blob/7002d53a079c06ab3a94f5485f022681a41dc9ed/protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/record/value/variable/VariableRecord.java#L34-L42

@berkaycanbc Does that clarify it?